v2/tools/asoctl/internal/importing/resource_import_result.go

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package importing
+
+import (
+	"bufio"
+	"io"
+	"os"
+
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/yaml"
+)
+
+// ResourceImportResult represents the result of an import operation
+type ResourceImportResult struct {
+	resources []genruntime.MetaObject
+}
+
+func (r *ResourceImportResult) SaveToWriter(destination io.Writer) error {
+	buf := bufio.NewWriter(destination)
+	defer func(buf *bufio.Writer) {
+		_ = buf.Flush()
```

Comment:
Is this actually best practice? Were you seeing issues without it?

Reply:
I've simplified to 

``` go
defer buf.Flush()
```

but I'm unsure if that's what you were concerned about.

Reply:
I was asking about the act of flushing at all, but I missed that this was `bufio` so it is needed.